src/main/kotlin/no/nav/syfo/cronjob/statusendring/PublishDialogmoteStatusEndringService.kt

```diff
@@ -32,13 +32,46 @@ class PublishDialogmoteStatusEndringService(
             pDialogmote = database.getDialogmote(id = moteId).first(),
             personIdent = database.getMoteDeltakerArbeidstaker(moteId).personIdent,
             virksomhetsnummer = database.getMoteDeltakerArbeidsgiver(moteId).virksomhetsnummer,
+            isStatusEndretForMotedeltakerBehandler = isRelatedToMotedeltakerBehandler(
+                dialogmoteStatusEndret = dialogmoteStatusEndret,
+            ),
         )
         dialogmoteStatusEndringProducer.sendDialogmoteStatusEndring(kDialogmoteStatusEndring)
 
         database.updateMoteStatusEndretPublishedAt(
             moteStatusEndretId = dialogmoteStatusEndret.id,
         )
     }
+
+    fun isRelatedToMotedeltakerBehandler(
+        dialogmoteStatusEndret: DialogmoteStatusEndret,
+    ): Boolean {
+        val moteId = dialogmoteStatusEndret.moteId
+        val behandler = database.getMoteDeltakerBehandler(moteId)
+        behandler?.let { pMotedeltakerBehandler ->
+
+            val statusEndretListByStatus = database.getMoteStatusEndretForMote(
+                moteId = moteId
+            ).filter {
+                it.status == dialogmoteStatusEndret.status
+            }
+            val behandlerVarselListByStatus = database.getMotedeltakerBehandlerVarselForMotedeltaker(
+                motedeltakerBehandlerId = pMotedeltakerBehandler.id,
+            ).filter {
+                it.varselType == MotedeltakerVarselType.valueOf(dialogmoteStatusEndret.status.name)
+            }
+
+            val dialogmoteStatusEndretIndex = statusEndretListByStatus.map { it.id }.indexOf(dialogmoteStatusEndret.id)
```

Comment:
Etter denne linja tror jeg bare dette trengs (med return lengre opp): 
(dialogmoteStatusEndretIndex >= 0) &&
                behandlerVarselListByStatus.getOrNull(dialogmoteStatusEndretIndex)?.let { behandlerVarsel ->
                    behandlerVarsel.varselType.name == dialogmoteStatusEndret.status.name
                } != null

Reply:
Eller bare:
return behandler?.let { pMotedeltakerBehandler ->
.....
            return dialogmoteStatusEndretIndex > 0 &&
                    behandlerVarselListByStatus.getOrNull(dialogmoteStatusEndretIndex)?.let { behandlerVarsel ->
                        behandlerVarsel.varselType.name == dialogmoteStatusEndret.status.name
                    } != null
        } ?: false

Men trenger noen enhetstester for å verifisere ;)